src/v/storage/mvlog/tests/active_segment_test.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "base/units.h"
+#include "model/tests/random_batch.h"
+#include "model/tests/randoms.h"
+#include "random/generators.h"
+#include "storage/directories.h"
+#include "storage/mvlog/versioned_log.h"
+#include "storage/ntp_config.h"
+#include "test_utils/gtest_utils.h"
+
+#include <seastar/core/seastar.hh>
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+
+using namespace storage::experimental::mvlog;
+using namespace std::literals::chrono_literals;
+
+class ActiveSegmentTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        ntp_ = model::random_ntp();
+        base_dir_ = get_test_directory();
+        storage::directories::initialize(base_dir_).get();
+    }
+
+    void TearDown() override {
+        log_->close().get();
+        log_.reset();
+        std::filesystem::remove_all(std::filesystem::path(base_dir_));
+    }
+    versioned_log* make_log(
+      std::optional<size_t> segment_bytes,
+      tristate<std::chrono::milliseconds> segment_ms) {
+        auto overrides
+          = std::make_unique<storage::ntp_config::default_overrides>();
+        overrides->segment_size = segment_bytes;
+        overrides->segment_ms = segment_ms;
+        const auto ntp = model::random_ntp();
+        storage::ntp_config cfg(ntp, base_dir_, std::move(overrides));
+        storage::directories::initialize(cfg.work_directory()).get();
+        log_ = std::make_unique<versioned_log>(std::move(cfg));
+        return log_.get();
+    }
+
+    ss::future<ss::circular_buffer<model::record_batch>>
```

Comment:
nit: For future proofing, maybe this should use the `version_log::segments_t` type (you'll have to make it `public`).

Reply:
This is a group of batches, not segments